src/plugin/regions/index.js

```diff
@@ -201,7 +201,7 @@ export default class RegionsPlugin {
             params = {...params, minLength: this.regionsMinLength};
         }
 
-        const region = new this.wavesurfer.Region(params, this.util, this.wavesurfer);
+        const region = new this.wavesurfer.Region(params, this.params, this.util, this.wavesurfer);
```

Comment:
Could we maybe just merge both params here instead of passing a new arg?
```suggestion
        const region = new this.wavesurfer.Region({
          formatTimeCallback: this.params.formatTimeCallback,
          edgeScrollWidth: this.params.edgeScrollWidth,
          ...params
        }, this.util, this.wavesurfer);
```

Reply:
That makes more sense than what I was suggesting.

Regarding `edgeScrollWidth`, I hadn't wanted to touch it, as I had an inkling that the code in the if block in `_onBackendCreated` was there to work around this issue, and I was a bit unsure what to make of it. Having looked at it some more, it seems to me that is what it's doing, but that the logic for setting a default value of `edgeScrollWidth` doesn't work unless all the regions are supplied in the plugin parameters - and I think it's worth sorting that out.

Would it be acceptable to have this in `_onBackendCreated`:
```
    if (!this.params.edgeScrollWidth) {
        this.params.edgeScrollWidth = this.wrapper.clientWidth * scrollWidthProportion;
    }
```
or should `this.params` be treated as read-only?

Reply:
Sounds like it's less trivial than just passing it from params.

For the sake of this PR (since you just wanted to fix `formatTimeCallback`), maybe let's skip `edgeScrollWidth` for now.

If you feel like tackling that, a new PR would be nice.

Reply:
Sorry, I just raised an issue for `edgeScrollWidth`, and then realised that PR is "pull request", not "performance report".